core/src/main/java/org/apache/james/mime4j/codec/DecoderUtil.java

```diff
@@ -284,6 +275,22 @@ private static String tryDecodeEncodedWord(
         }
     }
 
+    private static Charset lookupCharset(
```

Comment:
it looks much better, thank you.
I would be even more easier to read with early returns instead of variable re-allocation

Reply:
I tightened this as much as I dared.